src/Strategy/Conda/EnvironmentYml.hs

```diff
@@ -0,0 +1,94 @@
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Strategy.Conda.EnvironmentYml (
+  analyze',
+  buildGraph,
+  EnvironmentYmlFile(..)
+) where
+
+import Effect.Exec
+import Effect.ReadFS
+import qualified Data.Map.Strict as M
+import qualified Data.Text as T
+import Control.Carrier.Diagnostics hiding (fromMaybe)
+import Graphing (Graphing, fromList)
+import Data.Aeson
+import Data.Text (Text)
+import Data.Maybe (fromMaybe)
+import Data.List.Extra ((!?))
+import Path
+import Types
+
+
+buildGraph :: EnvironmentYmlFile -> Graphing Dependency
+buildGraph envYmlFile = Graphing.fromList (map toDependency allDeps)
+  where
+    allDeps = map getCondaDepFromText (dependencies envYmlFile)
+    toDependency :: CondaDep -> Dependency
+    toDependency CondaDep{..} =
+       Dependency 
+       {
+        dependencyType = CondaType,
+        dependencyName = depName,
+        dependencyVersion = CEq <$> depVersion, -- todo - properly handle version constraints
+        dependencyLocations = [],
+        dependencyEnvironments = [],
+        dependencyTags = M.empty
+      }
+
+analyze' :: 
+ ( Has ReadFS sig m
+ , Has Diagnostics sig m
+ )
+  => Path Abs File -> m (Graphing Dependency)
+analyze' envYml = buildGraph <$> readContentsYaml @EnvironmentYmlFile envYml
+
+
+-- an example Text: "biopython=1.78=py38haf1e3a3_0"
+-- '=' is a delimeter for <name>=<version>=<build>
+-- where <version> and <build> are optional
```

Comment:
[nit]: We use haddock-style comments where applicable.  [Here is the haddock documentation](https://haskell-haddock.readthedocs.io/en/latest/markup.html)